sparse/_compressed/convert.py

```diff
@@ -58,3 +63,250 @@ def uncompress_dimension(indptr):  # pragma: no cover
     for i in range(len(indptr) - 1):
         uncompressed[indptr[i] : indptr[i + 1]] = i
     return uncompressed
+
+
+@numba.jit(nopython=True, nogil=True)
+def is_sorted(arr):  # pragma: no cover
+    """
+    function to check if an indexing array is sorted without repeats. If it is,
+    we can use the faster slicing algorithm.
+    """
+    for i in range(len(arr) - 1):
+        if arr[i + 1] <= arr[i]:
+            return False
+    return True
+
+
+def _1d_reshape(x, shape, compressed_axes):
+    @numba.jit(nopython=True, nogil=True)
+    def _linearize(
```

Comment:
You want to either define this outside the function (if possible) or somehow cache it. The reason is that Numba/LLVM don't properly garbage collect this function once it goes out of memory.

Reply:
That's helpful. Thanks